### PR TITLE
Update for Solidus 2.1

### PR DESCRIPTION
**Pull request by kennyadsl**
This PR also fixes almost all deprecation warnings

ref #5 and solidusio/solidus#1687

### CODE REVIEW

**Comment by tvdeyen** [comment]
We should test if this still works for older solidus versions. Will provide a PR with a build matrix that makes testing against several solidus versions possible

**Comment by tvdeyen** [comment]
Intersting. The build setup does not test against Solidus 2.x, so we can't really rely on the green build.

**Comment by tvdeyen** [comment]
@kennyadsl please rebase your branch with master after #7 was merged and then update the `gemspec` and `Gemfile` to add support for Solidus 2.x

```ruby
# Gemfile
branch = ENV.fetch('SOLIDUS_BRANCH', 'master')
```

```ruby
# gemspec
s.add_dependency 'solidus', ['>= 1.0', '< 3']
```

and in the `.travis.yml` file add

```yml
    - SOLIDUS_BRANCH=v2.0 DB=mysql
    - SOLIDUS_BRANCH=v2.0 DB=postgresql
    - SOLIDUS_BRANCH=v2.1 DB=mysql
    - SOLIDUS_BRANCH=v2.1 DB=postgresql
    - SOLIDUS_BRANCH=master DB=mysql
    - SOLIDUS_BRANCH=master DB=postgresql
```

then we are sure to build for solidus >= 2.0 as well

**Comment by kennyadsl** [comment]
@tvdeyen thanks for the hints. I changed stuff as requested but there are still failures on trevis which I do not have in my local machine using the same ENV variables. Will take a deeper look tomorrow.

**Comment by tvdeyen** [comment]
@kennyadsl thanks. The build errors are related to Bundler 1.14.3. The whole 1.14 release series is a mess and broken in several ways. Please downgrade bundler on Travis like this:

```yml
# .travis.yml
before_install:
  - rvm use @global
  - gem uninstall bundler -x
  - gem install bundler --version=1.13.7
  - bundler --version
```

**Comment by kennyadsl** [comment]
Yes, just saw that on the solidus slack channel. Will provide a fix soon.

I'd also want to understand if solving the new Rails 5 deprecation warning for controller specs:

```
Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }
New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
```

will break old solidus versions builds. But maybe I can open a new PR for that.

**Comment by tvdeyen** [comment]
@kennyadsl no need to address the deprecation now. A separate PR is great.

**Comment by kennyadsl** [comment]
@tvdeyen I think we are ok now. The only failure was a PhantomJS crash but I don't have write access to this repo so I cannot rebuild it.

File: .travis.yml [hunk 1]
```diff
@@ -16,3 +16,14 @@ env:
     - SOLIDUS_BRANCH=v1.3 DB=postgresql
     - SOLIDUS_BRANCH=v1.4 DB=mysql
     - SOLIDUS_BRANCH=v1.4 DB=postgresql
```
Comment (tvdeyen) [inline .travis.yml]: We should remove all versions older than v1.4

File: solidus_reviews.gemspec [hunk 2]
```diff
@@ -17,19 +17,19 @@ Gem::Specification.new do |s|
   s.require_path = 'lib'
   s.requirements << 'none'
 
-  s.add_dependency 'solidus', '~> 1.0'
-  s.add_dependency 'deface', '~> 1.0'
-  s.add_dependency 'solidus_auth_devise', '~> 1.5'
+  s.add_dependency 'solidus', ['>= 1.0', '< 3']
```
Comment (tvdeyen) [inline solidus_reviews.gemspec]: We should use Solidus `>= 1.4`

**Comment by kennyadsl** [comment]
@tvdeyen do you mean icons in buttons?

**Comment by tvdeyen** [comment]
@kennyadsl yes 😊

**Comment by kennyadsl** [comment]
@tvdeyen I thought it was better to remove icons in all versions and keep compatibility with older ones. Do you think it can cause issue to have some links without icons in older versions?

**Comment by tvdeyen** [comment]
@kennyadsl not code-wise, no. But I like to keep the UI consistent. It's not necessary, but won't hurt either.

**Comment by kennyadsl** [comment]
@tvdeyen I totally understand the concern, just don't know which option is better. I'll remove those versions if you prefer this way 👍

**Comment by tvdeyen** [comment]
@kennyadsl remove them. There is v1.0.0 (which I just released) for Solidus < 1.4 
Will release 1.1 for Solidus >= 1.4 then

**Comment by kennyadsl** [comment]
@tvdeyen done!

**Pull request merged**